pkgs/build-support/node/fetch-bun-deps/bun-build-hook.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+bunBuildHook() {
+    runHook preBuild
+    echo "Executing bunBuildHook"
+
+    if [ -z "${bunBuildScript-}" ]; then
+        bunBuildScript="build"
+    fi
+
+    if ! type node > /dev/null 2>&1; then
+        echo "bunConfigHook WARNING: a node interpreter was not added to the build, and is probably required to run 'bun $bunBuildScript'. A common symptom of this is getting 'command not found' errors for Nodejs related tools."
+    fi
+
+    # shellcheck disable=SC2154
+    # SC2154: bunBuildFlags is referenced but not assigned
+    # This variable is provided by the Nix build environment, not assigned in this script
+    if [[ -n "${bunBuildFlags-}" ]]; then
+        bun run "$bunBuildScript" "${bunBuildFlags[@]}"
```

Comment:
```suggestion
        bun "$bunBuildScript" "${bunBuildFlags[@]}"
```

Bun actually doesn't need the run prefix and will look for scripts in your package json if it is not a built in command

i.e.
```
bun run xyz
```

is equal to

```
bun xyz
```

Reply:
No harm in using run, though right? pnpm behaves the same way, and I tend to include it out of habit. Just wondering, is omitting run the preferred style in Bun?

Reply:
I've seen both but I believe omitting it is preferred and run is included so bun can be aliases to node although I may be wrong